_posts/2020-05-18-websocket-fallback-sockjs.md

```diff
@@ -0,0 +1,329 @@
+---
+layout:    post
+title:     "WebSocket fallbacks - SockJS"
+date:      2020-05-18 08:00:00 +0100
+published: true
+author:    jgoszczurny
+image:     websocket_fallback_main_post_image.png
+tags:
+    - java
+    - spring
+    - javascript
+    - js
+    - websocket
+    - fallback
+    - sockjs
+    - atmosphere
+---
+
+# Wprowadzenie
+W ramach projektu udostępniającego ekran za pomocą przeglądarki z wykorzystaniem WebRTC, 
+zrobiliśmy moduł "signaling" który przekazuje komunikaty między użytkownikami.
+
+Najbardziej logicznym i wygodnym podejściem do tego jest wykorzystanie WebSocket. 
+W większości przypadków sprawdza się idealnie, chociaż może się zdarzyć że użytkownik 
+nie będzie miał możliwości ustanowienia połączenia WebSocket (np. nieskonfigurowane proxy, które nie przekazuje poprawnie komunikatów WebSocket, wycinając nagłówki), 
+z tego powodu konieczne było znalezienie sposobu, który to ominie.
+
+Początkowo w ramach prac badawczych, miał być zbadany Atmosphere, ponieważ pobieżnie był już wykorzystany w innym projekcie. 
+Wstępne zapoznanie z biblioteką pokazało że może ona zrobić WebSocket fallback za nas.
+
+
+# Biblioteki
+Podczas przeglądu literatury udało się również znaleźć bibliotekę SockJS,
+która wygląda na mniejszą i prostszą bibliotekę do zintegrowania z istniejącym projektem.
+
+
+## Atmosphere
+Jest to framework, który dostarcza komponenty dla klienta i serwera umożliwiające zbudowanie aplikacji webowej. 
+Jest możliwe wpięcie go w Spring, chociaż wymaga to więcej pracy do wykonania.
+
+Wspiera on transparentnie metody komunikacji takie jak:
+* WebSockets,
+* Server Sent Events (SSE),
+* Long-Polling,
+* HTTP Streaming (Forever frame)
+* JSONP
+
+Można uruchomić tą bibliotekę na serwerze, który wspiera servlety.
+
+## SockJS
+Biblioteka dekorująca WebSocket, dzięki czemu umożliwia fallback w przypadku gdy WebSocket nie jest dostępny.
+
+Wygodna w użyciu, ponieważ w JS działa ona tak samo jak zwykły WebSocket, 
```

Comment:
Jest wygodna w użyciu, ponieważ w JS działa tak samo jak zwykły WebSocket
==
odnośnie do tego określnika "zwykły", czy tym chcesz opisać jakieś poziomy WebSocketa czy chodzi o zaznaczenie że działa tak samo jak Websocket? Jeżeli drugi przypadek, to wówczas zrezygnowałbym z tego przecinka.